acousticbrainz/csv/ab_json2csv.py

```diff
@@ -0,0 +1,79 @@
+"""
+This script does not take any commandline arguments. It parses all the .JSON files from the
+../data folder into a single CSV file named output.csv.
+"""
+
+import os
+import json
+from fnmatch import fnmatch
+import pandas as pd
+
+# Load JSON data from a file
+JSON_FILES_PATH = "../data"
+PATTERN = "*.json"
+df_list = []
+
+
+def expand_lists(dataframe, list_columns):
+    """
+    Expand each column containing lists
+    """
+    for col in list_columns:
+        dataframe = dataframe.explode(col)
+    return dataframe
+
+
+for path, subdirs, json_files in os.walk(JSON_FILES_PATH):
+    for json_file in json_files:
+        with open(os.path.join(path, json_file), encoding="utf-8") as data_file:
+            if fnmatch(json_file, PATTERN):
+                data = json.load(data_file)
+
+        df = pd.json_normalize(data, max_level=2)
+        df_list.append(df)
+
+# Convert to CSV
+CSV_FILE = "output.csv"
+df_merged = pd.concat(df_list)
+df_merged = df_merged[
+    ["metadata.tags.musicbrainz_recordingid"]
+    + [
+        col
+        for col in df_merged.columns
+        if col != "metadata.tags.musicbrainz_recordingid"
+    ]
+]
+
+# Loop through columns and apply transformations
+columns_to_drop = []
+columns_with_lists = []
+
+for column in df_merged.columns:
+    if (
+        df_merged[column].apply(lambda x: isinstance(x, dict)).any()
+        or df_merged[column].apply(lambda x: isinstance(x, float)).any()
+    ) and not column.endswith("id"):
+        # Mark column for dropping if any element is a dictionary or float
+        columns_to_drop.append(column)
+    elif df_merged[column].apply(lambda x: isinstance(x, list)).any():
+        # Convert lists to strings
+        columns_with_lists.append(column)
```

Comment:
Why are these applied?

Reply:
I don't really understand the question.

Reply:
Why do you drop any column with a float (except those ending with "id")? Why do you drop any column with a `dict`?

Reply:
The floats are the percentages for the genre features extraction. Those data are too overwhelming and makes my inspection much more difficult. Sometimes python sees an ID as a float, so I want to keep those. If we normalize the higher level dictionaries, they makes the chart very large, and if we not normalize the higher level dictionaries, they will be left as dictionaries in the CSV, so not good as well. Therefore I decide to drop any dictionaries after level 2.

Reply:
Ok...make a comment! These are exactly the kinds of things that you need to explain, otherwise it looks like it was random.

Reply:
I don't understand what you mean by "overwhelming." I'll be in tomorrow so that you can explain this to me in person, then you can clarify that here afterwards.

Reply:
It means too much decimal numbers and hard to inspect.

Reply:
What do you mean too much decimal numbers? For what? For whom is it hard to inspect? What are we inspecting?

Reply:
The decimal numbers are for the percentages of the genre features extraction. Since I am at testing stage and I have to inspect the data myself, leaving too much numbers makes the CSV very long and hard to read.

Reply:
I still don't understand what the issue is. We'll talk tomorrow.

Reply:
Ok, we talked. @Yueqiao12Zhang was dropping some columns because he was just testing the script. He will include all columns at the end.